src/main.py

```diff
@@ -336,7 +336,7 @@ def initialize_warehouse_connection(ctx: Context, connection_name: str) -> dict:
 
 @mcp.tool()
 @track
-def run_query(ctx: Context, query: str) -> pd.DataFrame:
+def run_query(ctx: Context, query: str) -> dict:
```

Comment:
This was done to fix `TypeError: Object of type DataFrame is not JSON serializable` because Redshift may return JSON object as JSONB/JSON also. Now it's unified for all warehouses.